plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java

```diff
@@ -102,16 +110,30 @@ public void execute(String sql)
             }
         }
 
-        public void createDataset(String dataset)
+        public void createDataset(String datasetName)
         {
-            bigQuery.create(DatasetInfo.newBuilder(dataset).build());
+            DatasetInfo dataset = DatasetInfo.newBuilder(datasetName)
+                    .setLabels(ImmutableMap.copyOf(ImmutableSet.of(BIG_QUERY_SQL_EXECUTOR_LABEL)))
+                    .build();
+            bigQuery.create(dataset);
         }
 
         public void dropDataset(String dataset)
         {
             bigQuery.delete(dataset);
         }
 
+        public void deleteSelfCreatedDatasets()
```

Comment:
nit: maybe `deleteTestDatasets`?

Reply:
Good point. Is there some reason why datasets get created without calling `createDataset` though? Maybe we can clean that up in the future?

Let's keep as is for now.